patches/chromium/feat_add_data_parameter_to_processsingleton.patch

```diff
@@ -0,0 +1,333 @@
+@@ -563,6 +563,7 @@ class ProcessSingleton::LinuxWatcher
+   // |reader| is for sending back ACK message.
+   void HandleMessage(const std::string& current_dir,
+                      const std::vector<std::string>& argv,
++                     const base::span<const uint8_t>& additional_data,
+                      SocketReader* reader);
+ 
+  private:
+@@ -619,13 +620,16 @@ void ProcessSingleton::LinuxWatcher::StartListening(int socket) {
+ }
+ 
+ void ProcessSingleton::LinuxWatcher::HandleMessage(
+-    const std::string& current_dir, const std::vector<std::string>& argv,
++    const std::string& current_dir,
++    const std::vector<std::string>& argv,
++    const base::span<const uint8_t>& additional_data,
+     SocketReader* reader) {
+   DCHECK(ui_task_runner_->BelongsToCurrentThread());
+   DCHECK(reader);
+ 
+   if (parent_->notification_callback_.Run(base::CommandLine(argv),
+-                                          base::FilePath(current_dir))) {
++                                          base::FilePath(current_dir),
++                                          additional_data)) {
+     // Send back "ACK" message to prevent the client process from starting up.
+     reader->FinishWithACK(kACKToken, base::size(kACKToken) - 1);
+   } else {
+@@ -673,7 +677,8 @@ void ProcessSingleton::LinuxWatcher::SocketReader::
+     }
+   }
+ 
+-  // Validate the message.  The shortest message is kStartToken\0x\0x
++  // Validate the message.  The shortest message kStartToken\0\00
++  // The shortest message with additional data is kStartToken\0\00\00\0.
+   const size_t kMinMessageLength = base::size(kStartToken) + 4;
+   if (bytes_read_ < kMinMessageLength) {
+     buf_[bytes_read_] = 0;
+@@ -698,15 +703,29 @@ void ProcessSingleton::LinuxWatcher::SocketReader::
+   timer_.Stop();
+ 
+   std::string current_dir = tokens[1];
++#if 0
+   // Remove the first two tokens.  The remaining tokens should be the command
+   // line argv array.
+   tokens.erase(tokens.begin());
+   tokens.erase(tokens.begin());
++#endif
++
++  size_t num_args;
++  base::StringToSizeT(tokens[2], &num_args);
++  std::vector<std::string> command_line(tokens.begin() + 3, tokens.begin() + 3 + num_args);
++
++  base::span<const uint8_t> additional_data;
++  if (tokens.size() == 5 + num_args) {
++    size_t additional_data_size;
++    base::StringToSizeT(tokens[3 + num_args], &additional_data_size);
++    const uint8_t* additional_data_bits = reinterpret_cast<const uint8_t*>(tokens[4 + num_args].c_str());
++    additional_data = base::span<const uint8_t>(additional_data_bits, additional_data_bits + additional_data_size);
++  }
+ 
+   // Return to the UI thread to handle opening a new browser tab.
+   ui_task_runner_->PostTask(
+       FROM_HERE, base::BindOnce(&ProcessSingleton::LinuxWatcher::HandleMessage,
+-                                parent_, current_dir, tokens, this));
++                                parent_, current_dir, command_line, additional_data, this));
```

Comment:
Oops, this is being passed to another sequence. The previous implementation of passing the vector directly was correct here.